lib/test.js

```diff
@@ -48,3 +58,23 @@ Test.prototype.clone = function() {
   test.ctx = this.ctx;
   return test;
 };
+
+/**
+ * {@link Test}-related constants.
+ * @public
+ * @memberof Test
+ * @readonly
+ * @enum {string}
+ */
+var constants = {
+  /**
+   * Value of {@link Test#state} when a `Test` has failed
+   */
+  STATE_FAILED: 'failed',
+  /**
+   * Value of {@link Test#state} when a `Test` has passed
+   */
+  STATE_PASSED: 'passed'
+};
```

Comment:
Are there other states?
* `STATE_INIT` (maybe `null` or `undefined`)
* `STATE_PENDING`?

Reply:
unfortunately, `pending` is not a state, it's a prop.  it *should* be a state.  there are no other states.

Reply:
To me, the code gets messy here, having other states possible but not accounted for. Think both of them would be good to describe states when not (yet) pass/fail.

Reply:
I'm not going to refactor the pending stuff in this PR, but I can add an initial state.

- [ ] add "initial" state

Reply:
this would necessarily replace the `undefined` business

Reply:
I was only in reference to adding the pending state constant `XXX.state = STATE_PENDING` when `XXX.pending === true`, not changing the pending processing logic. But if that's hard, let's do it in a different PR...

Reply:
right, yes, generally nothing is easy.

Reply:
*punting*